src/domain_bridge/wait_for_qos_handler.hpp

```diff
@@ -0,0 +1,206 @@
+// Copyright 2021, Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DOMAIN_BRIDGE__WAIT_FOR_QOS_HANDLER_HPP_
+#define DOMAIN_BRIDGE__WAIT_FOR_QOS_HANDLER_HPP_
+
+// Silly cpplint thinks this is a C system header
+#include <optional>
+
+#include <atomic>
+#include <functional>
+#include <memory>
+#include <string>
+#include <thread>
+#include <vector>
+
+#include "rclcpp/node.hpp"
+#include "rclcpp/qos.hpp"
+
+namespace domain_bridge
+{
+
+/// QoS match information.
+/**
+ * Used as a return value for QoS matching functions.
+ */
+struct QosMatchInfo
+{
+  explicit QosMatchInfo(const rclcpp::QoS & init_qos)
+  : qos(init_qos)
+  {}
```

Comment:
```suggestion
  explicit QosMatchInfo()
  : qos{10}
  {}
```

see next comment.
(depth doesn't matter, I guess it's overriden by the user provided depth (if there is one).